transform/coroutines.go

```diff
@@ -763,6 +763,27 @@ func (c *coroutineLoweringPass) lowerCallReturn(caller *asyncFunc, call llvm.Val
 // lowerFuncCoro transforms an async function into a coroutine by lowering async operations to `llvm.coro` intrinsics.
 // See https://llvm.org/docs/Coroutines.html for more information on these intrinsics.
 func (c *coroutineLoweringPass) lowerFuncCoro(fn *asyncFunc) {
+	// Ensure that any alloca instructions in the entry block are at the start.
+	// Otherwise, block splitting would result in unintended behavior.
+	{
+		// Skip alloca instructions at the start of the block.
+		inst := fn.fn.FirstBasicBlock().FirstInstruction()
+		for !inst.IsAAllocaInst().IsNil() {
+			inst = llvm.NextInstruction(inst)
+		}
```

Comment:
I believe this won't work correctly if there are no alloca instructions in the entry block.

Reply:
The loop condition is the other way around. If there are no alloca instructions this will run 0 times.

Reply:
Oh, I see. I got confused by the double negation.